### PR TITLE
[Merged by Bors] - feat(Algebra/GroupPower): weaken TC assumptions

### PR DESCRIPTION
**Pull request by urkud**
- Assume `OrderedSemiring` instead of `StrictOrderedSemiring`
  here and there.
- Add `@[simp]` to `zsmul_one`.
- Add `exists_lt_nsmul`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alexjbest** [comment]
maintainer merge

File: Mathlib/Algebra/GroupPower/Lemmas.lean
Comment (alexjbest) [inline Mathlib/Algebra/GroupPower/Lemmas.lean:654]: did the ac_rfl stop working? it seems a more principled and maintainable proof than simp only to me

File: Mathlib/Algebra/GroupPower/Lemmas.lean
Comment (eric-wieser) [inline Mathlib/Algebra/GroupPower/Lemmas.lean:636]: I'm a little wary of this one: didn't we decide `OrderedSemiring` was nonsense? Maybe @YaelDillies remembers the history better.

Comment (Vierkantor) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:636]: I believe Yaël fixed the weirdness a while ago: [mathlib#16172](https://github.com/leanprover-community/mathlib/pull/16172).

**Review by Vierkantor — APPROVED** [review]
LGTM with Alex's comments. I'll let Eric decide if `OrderedSemiring` is non-nonsense enough to get this merged :)

bors d=@eric-wieser

File: Mathlib/Algebra/GroupPower/Lemmas.lean
Comment (Vierkantor) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:636]: By the way, Yaël is on vacation this week, according to Zulip. So I don't want to block on them :)

**Comment by eric-wieser** [comment]
Thanks for the archaeology @Vierkantor, I'm convinced!

bors d+

(we're waiting to address Alex's comment still, right?)

File: Mathlib/Algebra/GroupPower/Lemmas.lean
Comment (urkud) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:654]: Yes, it didn't work. I don't know why.

Comment (urkud) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:654]: @alexjbest Do you want to have a look, or should I merge it?

Comment (alexjbest) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:654]: I'd like to understand why but it shouldn't hold up this PR! so please merge and I can investigate later

**Comment by urkud** [comment]
bors merge

**Comment by urkud** [comment]
Both failures are because of `lake exe cache get`
bors merge

File: Mathlib/Algebra/GroupPower/Lemmas.lean
Comment (alexjbest) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:654]: minimised and reported as lean4#2413

File: Mathlib/Algebra/GroupPower/Lemmas.lean
Comment (YaelDillies) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:636]: Just came back from my holidays. Anne is perfectly right!

**Pull request closed without merging**
